src/core/lavfaudio.cpp

```diff
@@ -66,8 +84,21 @@ FFLAVFAudio::FFLAVFAudio(const char *SourceFile, int Track, FFMS_Index &Index, i
 {
 	LAVFOpenFile(SourceFile, FormatContext, TrackNumber);
 
-	CodecContext.reset(FormatContext->streams[TrackNumber]->codec);
-	assert(CodecContext);
+	AVCodec *Codec = avcodec_find_decoder(FormatContext->streams[TrackNumber]->FFMSCODEC->codec_id);
```

Comment:
This stuff seems worth pulling out into a function rather than duplicating (or maybe just rename `ReopenFile()` to `OpenFile()` and make it handle the case where the file isn't already open).